app/helpers/toolbar_helper.rb

```diff
@@ -191,7 +191,7 @@ def toolbar_button_normal(props, view_button = false)
   # attributes in html
   #
   def data_hash_keys(props)
-    hsh = %i(pressed popup console_url prompt explorer onwhen url_parms url).each_with_object({}) do |key, h|
+    hsh = %i(pressed popup window_url prompt explorer onwhen url_parms url).each_with_object({}) do |key, h|
       h["data-#{key}"] = props[key] if props.key?(key)
     end
     hsh
```

Comment:
I might be wrong here, but as far as I remember `content_tag`, `button_tag` and other Rails view helpers can accept a bag of data attributes as a hash. In other words, there is no need to convert the attributes to `{ "data-xxx" => "something", "data-yyy" => "else" }` format, passing it as `{ "data" => {"xxx" => "something", "yyy" => "else" } }` will work just fine.

It means that `data_hash_keys` can be defined as

```ruby
      def data_hash_keys(props)
        {
           data: props.slice(*%i(pressed popup window_url prompt explorer onwhen url_parms url))
        }
      end
```

or something like this.